ncm-accounts/src/main/perl/accounts.pm

```diff
@@ -390,14 +390,28 @@ sub apply_profile_groups
     my ($self, $system, $profile) = @_;
 
     while (my ($group, $cfg) = each(%$profile)) {
+      my $required_members = $cfg->{requiredMembers};
+      $required_members = [] unless $required_members;
       if (!exists($system->{groups}->{$group})) {
         $self->debug(2, "Scheduling addition of group $group");
         $system->{groups}->{$group} = { name => $group,
-                                        members => {},
+                                        members => {map(($_ => 1), @$required_members)},
                                         gid => $cfg->{gid}};
-      } elsif ($system->{groups}->{$group}->{gid} != $cfg->{gid}) {
-        $self->debug(2, "Changing gid of group $group to $cfg->{gid}");
-        $system->{groups}->{$group}->{gid} = $cfg->{gid};
+      } else {
+        if ($system->{groups}->{$group}->{gid} != $cfg->{gid}) {
+          $self->debug(2, "Changing gid of group $group to $cfg->{gid}");
+          $system->{groups}->{$group}->{gid} = $cfg->{gid};
+        }
+        if ( $cfg->{replaceMembers} ) {
+          $self->debug(2, "Group $group: replacing existing member list");
+          $system->{groups}->{$group}->{members} = {};
+        }
+        if ( @$required_members ) {
+          $self->debug(3, "Group $group: adding required members (",join(',',@$required_members),")");
```

Comment:
I like to see components summarise their effect at info level ("Changed N groups", "Added Y users") and then report individual actions at verbose level ("adding members X,Y,Z", "replacing users with ...") so I would definitely like to see these as verbose. I agree info is probably too much but too many components don't give any indication of what they have changed unless run in debug mode!